### PR TITLE
fix: Re-enable webfinger feature

### PR DESCRIPTION
**Pull request by roschaefer**
Ok, so here is the plan. Let's give both our cucumber features and your
cypress tests a prominent place to live. That would be the root level
folder of our application. Second, let's revive formerly dead code step
by step.

Ie. move code from the former location `backend/features/` to `features/`
when it is ready. All edge cases should be tested with unit tests in
`backend/`, see my `webfinger.spec.js` as an example.

### Issues

- None

### Todo

- [X] None

### CODE REVIEW

**Comment by datenbrei** [comment]
"WebFinger is a protocol specified by the Internet Engineering Task Force IETF that allows for discovery of information about people and things identified by a URI.[1] Information about a person might be discovered via an "acct:" URI, for example, which is a URI that looks like an email address."

@roschaefer What data privacy concerns do we have to address here?

File: .travis.yml [hunk 1]
```diff
@@ -8,13 +8,13 @@ addons:
...
 
-before_install:
+install:
   - yarn global add wait-on
   # Install Codecov
   - yarn install
   - cp cypress.env.template.json cypress.env.json
 
-install:
+before_script:
```
Comment (mattwr18) [inline .travis.yml]: how is this and the change :point_up_2: related to this `PR`? 
is this some setup that should be changed anyway, ie code creep, or is this somehow related to running the cucumber tests?

File: backend/src/activitypub/routes/webfinger.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        type: 'application/activity+json',
+        href: `${CONFIG.CLIENT_URI}/activitypub/users/${name}`,
+      },
+    ],
+  }
+}
+
+export async function handler(req, res) {
+  const { resource = '' } = req.query
+  // eslint-disable-next-line no-unused-vars
```
Comment (mattwr18) [inline backend/src/activitypub/routes/webfinger.js]: why would this fail lint? you are using `name` and `domain` on line 26??

File: backend/src/activitypub/routes/webfinger.spec.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
+import { handler } from './webfinger'
+import Factory from '../../seed/factories'
+import { getDriver } from '../../bootstrap/neo4j'
+
+let resource
+let res
+let json
+let status
+let contentType
```
Comment (mattwr18) [inline backend/src/activitypub/routes/webfinger.spec.js]: Is there any reason you prefer multi-line `let` statements over one-liners?
Just a question...

File: backend/src/activitypub/routes/webfinger.spec.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    it('sends HTTP 400', async () => {
+      await request()
+      expect(status).toHaveBeenCalledWith(400)
+      expect(json).toHaveBeenCalledWith({
+        error: 'Query parameter "?resource=acct:<USER>@<DOMAIN>" is missing.',
+      })
+    })
+  })
+
+  describe('?ressource query param', () => {
```
Comment (mattwr18) [inline backend/src/activitypub/routes/webfinger.spec.js]:
```suggestion
  describe('?resource query param', () => {
```

File: backend/src/activitypub/routes/webfinger.spec.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        })
+      })
+    })
+
+    describe('with acct:', () => {
+      beforeEach(() => {
+        resource = 'acct:some-user@domain'
+      })
+
+      it('returns empty json', async () => {
```
Comment (mattwr18) [inline backend/src/activitypub/routes/webfinger.spec.js]: seems to return an error still, oder?

File: backend/src/activitypub/routes/webfinger.spec.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+          expect(json).toHaveBeenCalledWith({
+            links: [
+              {
+                href: 'http://localhost:3000/activitypub/users/some-user',
+                rel: 'self',
+                type: 'application/activity+json',
+              },
+            ],
+            subject: 'acct:some-user@localhost:3000',
+          })
```
Comment (mattwr18) [inline backend/src/activitypub/routes/webfinger.spec.js]: really exciting stuff!!!

File: backend/src/activitypub/routes/webfinger.js (hunk 2 above)
Comment (roschaefer) [reply backend/src/activitypub/routes/webfinger.js]: It's the `_` that's being unused.

File: backend/src/activitypub/routes/webfinger.spec.js (hunk 3 above)
Comment (roschaefer) [reply backend/src/activitypub/routes/webfinger.spec.js]: No, I think a linter with a previous configuration complained about it.

File: backend/src/activitypub/routes/webfinger.spec.js (hunk 5 above)
Comment (roschaefer) [reply backend/src/activitypub/routes/webfinger.spec.js]: Genau

File: .travis.yml (hunk 1 above)
Comment (roschaefer) [reply .travis.yml]: Code creep, but a mild one. Starting the docker containers is more a `before_script` whereas installing e.g. `cucumber` is part of the `install` in my eyes.

File: backend/src/activitypub/routes/webfinger.js (hunk 2 above)
Comment (mattwr18) [reply backend/src/activitypub/routes/webfinger.js]: that shouldn't fail linting... it's kind of the purpose to use the `_` in this way
